### DISCUSSION TITLE
Possible to treat a _single_ internal link as external?

### DISCUSSION THREAD

**Discussion by minrk** [executablebooks/MyST-Parser#677] [Q&A] [answered]
Short summary: I want to link to an anchor on an internal [page](https://jupyterhub.readthedocs.io/en/3.1.0/reference/rest-api.html) that _I_ know will be there, but myst can't reasonably know will be there. I believe `myst_all_links_external=True` would work for this, but I don't want to do that for _all_ links, just this one.

Essentially, I want to emit [this link](https://jupyterhub.readthedocs.io/en/3.1.0/reference/rest-api.html#/default/get_users).

More detail:

I have a [page](https://github.com/jupyterhub/jupyterhub/blob/3.1.0/docs/source/reference/rest-api.md?plain=1) where some of the content is loaded via javascript (openapi), so the anchors are available to myst.

I want to create a link to this page, with an anchor that myst doesn't know about, and let myst 'trust me' that the link will resolve. I've tried e.g.

```markdown
[md](./rest-api.md#/default/get_users)
[html](./rest-api.html#/default/get_users)
```

and it fails with

```
WARNING: 'myst' reference target not found: ./rest-api.html#/default/get_users
```

I'd be happy with the warning (makes sense, myst doesn't know about it!) _if_ it still created the warned-about link, but because the target doesn't resolve, no link is added.

Is there a way without enabling _all_ external link handling to get one unresolved anchor link to work?

**Comment by lesteve** [comment]
I had a similar use case in https://github.com/pyodide/pyodide/pull/3244#issuecomment-1310310791, glad to see I am not the only one!

I naively thought using HTML directly would fix it but this doesn't work, probably because MyST is parsing HTML in a clever way.

```html
<a href="../console.html">attempt with external</a>
```

Adding the external class makes it work e.g. but I am not sure how robust this is (I also used class reference because this is what is used for links in the generated HTML but I don't think this is crucial):
```html
<a class="reference external" href="../console.html">attempt with external</a>
```

In our case we ended ups using the `eval-rst` directive and write the link in rST. This is cumbersome but we thought this was the most robust solution:

**Comment by chrisjsewell** [comment, marked as answer]
Heya @minrk see https://myst-parser.readthedocs.io/en/stable/syntax/cross-referencing.html#customising-external-url-resolution

you can now use `[my-external-link](my-external-link){.external}`

**Reply by minrk** [reply]
Thanks, that [works](https://github.com/jupyterhub/jupyterhub/pull/4386)!
